gluten-core/src/main/scala/org/apache/gluten/extension/StrategyOverrides.scala

```diff
@@ -1,213 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.gluten.extension
-
-import org.apache.gluten.{GlutenConfig, GlutenSparkExtensionsInjector}
-import org.apache.gluten.backendsapi.BackendsApiManager
-import org.apache.gluten.extension.columnar.TRANSFORM_UNSUPPORTED
-import org.apache.gluten.extension.columnar.TransformHints.TAG
-import org.apache.gluten.utils.LogicalPlanSelector
-
-import org.apache.spark.sql.{SparkSession, SparkSessionExtensions, Strategy}
-import org.apache.spark.sql.catalyst.SQLConfHelper
-import org.apache.spark.sql.catalyst.expressions.Expression
-import org.apache.spark.sql.catalyst.optimizer.{BuildLeft, BuildRight, JoinSelectionHelper}
-import org.apache.spark.sql.catalyst.plans._
-import org.apache.spark.sql.catalyst.plans.logical._
-import org.apache.spark.sql.execution.{joins, JoinSelectionShim, SparkPlan}
-import org.apache.spark.sql.execution.adaptive.{BroadcastQueryStageExec, LogicalQueryStage}
-import org.apache.spark.sql.execution.joins.BroadcastHashJoinExec
-
-object StrategyOverrides extends GlutenSparkExtensionsInjector {
-  override def inject(extensions: SparkSessionExtensions): Unit = {
-    extensions.injectPlannerStrategy(JoinSelectionOverrides)
-  }
-}
-
-case class JoinSelectionOverrides(session: SparkSession)
```

Comment:
Do we really to refactor strategy to columnar ? Since we have a cost evalutor and the plan tag can always pass to columnar rules, can we add a tag to specify the shj is converted by gluten and then we can fallback it to smj at columnar rule side if it can not be offloaded.

Reply:
Adding a tag only puts the buildSide selection in front, which seems to make little difference.

Reply:
If possible, we'd try our best to offload a computation in one single place in code. If we use tag in strategy and read that tag in columnar rule for fallback, that would mean the e.g. shj plan can be barely allowed to change between the two operations. Such code would dramatically increase the planner/rule code complexity over time.

So I am inclined to doing join selection either in columnar rule or in strategy. Don't do it in both places as long as we have other options.